client.go

```diff
@@ -179,3 +179,34 @@ func (c *Client) getErrorFromResponse(resp *http.Response) (*errorObject, error)
 	}
 	return &s, nil
 }
+
+func (c *Client) pagedGet(basePath string,
+	handleResponse func(response *http.Response) (*APIListObject, error)) error {
+	// Indicates whether there are still additional pages associated with request.
+	var stillMore bool
+
+	// Offset to set for the next page request.
+	var nextOffset uint
+
+	// While there are more pages, keep adjusting the offset to get all results.
+	for stillMore, nextOffset = true, 0; stillMore; {
+		response, err := c.do("GET", fmt.Sprintf("%v?offset=%d", basePath, nextOffset), nil, nil)
+		if err != nil {
+			return err
+		}
+		defer response.Body.Close()
```

Comment:
Hm, that is a good point. As an alternative, my understanding is I could put `response.Body.Close()` after `handler(response)`, but that this may prevent response bodies from being closed in the event of panics caused by `handler`.

Reply:
Yeah, [confirmed](https://blog.golang.org/defer-panic-and-recover), and I think `defer` in general is just a much better way to handle these kinds of situations. Would like to talk about this.

Reply:
Ah, so... Interesting thing to note here. There is a function `decodeJSON` that defers the response body. This is what the `ListXYZ` methods in this package use. So, this `defer response.Body.Close` is actually unnecessary.

Reply:
Talk in person, we agreed that the best thing to do at this point is to document that `responseHandler` is responsible for closing the response.